vscode/DscExt/src/extension.ts

```diff
@@ -0,0 +1,67 @@
+/*---------------------------------------------------------
+ * Copyright (C) Microsoft Corporation. All rights reserved.
+ *--------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import * as fs from 'fs';
+
+export function activate(context: vscode.ExtensionContext) {
+
+	const dsc_type_provider = vscode.languages.registerCompletionItemProvider({ language: 'yaml', pattern: '**/*.dsc.yaml' },
+		{
+			provideCompletionItems(document: vscode.TextDocument, position: vscode.Position) {
+
+				const linePrefix = document.lineAt(position).text.slice(0, position.character);
+				if (!linePrefix.endsWith('type:')) {
+					return undefined;
+				}
+
+				let ps_cache_file_path: string = '';
+				if (process.platform==='win32')
+				{
+					ps_cache_file_path = process.env.LocalAppData + '\\dsc\\PSAdapterCache.json';
+				}
+				else
+				{
+					ps_cache_file_path = process.env.HOME + '/.dsc/PSAdapterCache.json';
+				}
+				console.log("Using cache path " + ps_cache_file_path);
+
+				if (!fs.existsSync(ps_cache_file_path)) {
+					console.log("Cache file does not exist");
+					return [];
+				}
+
+				const dataArray = JSON.parse(fs.readFileSync(ps_cache_file_path, 'utf-8'));
+				var completions:vscode.CompletionItem[] = [];
+				dataArray.ResourceCache.forEach(function (resouce: any) {
+						
+					const all_props_completion = new vscode.CompletionItem(resouce.Type);
+					const key_props_completion = new vscode.CompletionItem(resouce.Type + " [keys only]");
+
+					let all_props_comp_text: string = ' ' + resouce.Type;
+					let key_props_comp_text: string = ' ' + resouce.Type;
+					all_props_comp_text += '\nproperties:';
+					key_props_comp_text += '\nproperties:';
+					resouce.DscResourceInfo.Properties.forEach(function (prop: any) {
+						all_props_comp_text += '\n  ' + prop.Name + ': ';
```

Comment:
If you want, it's probably a lot more readable to write this snippet as multiple strings ("which will be joined as multiple lines upon insertion") and is there any insertions or variables we might want to provide inside the completion snippet? https://code.visualstudio.com/docs/editor/userdefinedsnippets#_create-your-own-snippets